src/widgets-service.js

```diff
@@ -79,15 +79,21 @@ WidgetsService.prototype.createCategoryModel = function (attrs, layer, state) {
  */
 WidgetsService.prototype.createHistogramModel = function (attrs, layer, state, opts) {
   _checkProperties(attrs, ['title']);
-  var dataAttrs = _.extend(attrs, state, {hasInitialState: this._widgetsCollection.hasInitialState()}); // Will overwrite preset attributes with the ones passed on the state
+  var dataAttrs = _.extend(attrs, state, { hasInitialState: this._widgetsCollection.hasInitialState() }); // Will overwrite preset attributes with the ones passed on the state
   var dataviewModel = this._dataviews.createHistogramModel(layer, dataAttrs);
 
-  var ATTRS_NAMES = ['id', 'title', 'order', 'collapsed', 'bins', 'show_stats', 'show_source', 'normalized', 'style', 'hasInitialState'];
-  var widgetAttrs = _.pick(attrs, ATTRS_NAMES);
+  // Default bins attribute was removed from dataViewModel because of time-series aggregation.
+  // Just in case it's needed for histogram models we added it here.
+  if (!dataviewModel.has('bins')) {
+    dataviewModel.set('bins', 10, { silent: true });
+  }
+
+  var attrsNames = ['id', 'title', 'order', 'collapsed', 'bins', 'show_stats', 'show_source', 'normalized', 'style', 'hasInitialState'];
```

Comment:
what about having a `._getAttrsNames`, for example where we pick the attrs depending on the type, instead of repeating the attrs on every create model?